evmlib/src/contract/payment_vault/interface.rs

```diff
@@ -44,3 +54,13 @@ impl From<QuotingMetrics> for IPaymentVault::QuotingMetrics {
         }
     }
 }
+
+fn data_type_conversion(data_type: u32) -> u8 {
+    match data_type {
+        0 => 2, // Chunk
+        1 => 0, // GraphEntry
+        2 => 3, // Pointer
+        3 => 1, // Scratchpad
+        _ => 4, // Does not exist
+    }
```

Comment:
Ouch, we already have such a conversion upstream from DataType to u32, this feels a bit awkward.

Reply:
Not blocking in any way though we can always fix that later.